pyinfra/facts/deb.py

```diff
@@ -44,7 +44,7 @@ class DebPackage(FactBase):
     requires_command = 'dpkg'
 
     def command(self, name):
-        return '! test -e {0} || (dpkg -I {0} 2> /dev/null || dpkg -s {0})'.format(name)
+        return '(test -e {0} && dpkg -I {0} 2> /dev/null) || (! test -e {0} && dpkg -s {0} 2>/dev/null) || true'.format(name)
```

Comment:
My concern here is with the final `|| true` which will capture any unexpected errors elsewhere in the command - if we flip the logic and use the following it should fix the bug without the issue:

```suggestion
        return '! test -e {0} && (dpkg -s {0} 2>/dev/null || true) || dpkg -I {0}'.format(name)
```